### PR TITLE
Dispatch an event if there is an exception

### PR DESCRIPTION
**Pull request by mpchadwick**
Allows others to hook in if there are other requirements (e.g. alerting system).

Based on discussions in #4

### CODE REVIEW

**Comment by mpchadwick** [comment]
@LeeSaferite Any thoughts on this?

**Comment by sprankhub** [comment]
:+1:

**Comment by kesonno** [comment]
+1 for this addition 👍

**Comment by LeeSaferite** [comment]
I see no reason not to merge this, but I no longer work for AOE and not longer have merge permissions.

@fbrnc

File: app/code/community/Aoe/DbRetry/Resource/Db/Pdo/Mysql/Adapter.php [hunk 1]
```diff
@@ -35,6 +35,7 @@ public function query($sql, $bind = [])
                     $try = $maxTries;
                 } catch (Exception $e) {
                     $try++;
+                    Mage::dispatchEvent('aoe_dbretry_exception', ['try' => $try, 'exception' => $e]);
```
Comment (LeeSaferite) [inline app/code/community/Aoe/DbRetry/Resource/Db/Pdo/Mysql/Adapter.php:38]: I think my only input here would be to document the event in the readme and be VERY clear that further DB calls in the event handler are unsupported and dangerous.

**Review by LeeSaferite — CHANGES_REQUESTED** [review]

**Pull request merged**